src/Slist.hs

```diff
@@ -490,6 +491,22 @@ cycle sl@(Slist _ Infinity) = sl
 cycle Slist{..}             = infiniteSlist $ L.cycle sList
 {-# INLINE cycle #-}
 
+{- | @O(1)@. A lazy slist equivalent of Haskell's '[from..to]' notation:
+creates an slist of sequentially ordered values starting at 'from' and ending at 'to'.
+
+>>> fromRange 0 5
+Slist {sList = [0,1,2,3,4,5], sSize = Size 6}
+>>> fromRange 5 0
+Slist {sList = [], sSize = Size 0}
+>>> fromRange 'a' 'd'
+Slist {sList = "abcd", sSize = Size 4}
+-}
+fromRange :: Enum a => a -> a -> Slist a
+fromRange from to
+  | fromEnum from <= fromEnum to = one from <> fromRange (succ from) to
```

Comment:
`succ` is not a typesafe function and it will fail on `maxBounds`.

Reply:
Ah, my mistake. For some reason I actually thought `succ` was cyclic (i.e. `succ True = False`). Although even then, the code is invalid for any types isomorphic to `()`.